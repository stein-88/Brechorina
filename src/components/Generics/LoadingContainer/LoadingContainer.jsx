import React from 'react'
import loadingImg from '@images/loading.svg'
import s from './LoadingContainer.scss'

const LoadingContainer = ({ height100, customPadding, onLoad }) => (
    <section className={`w-100 ${height100 ? s.heigh100 : 'h-auto'} text-center ${customPadding || 'pt-5 mt-5'}`}>
        <object
            alt="Loading"
            aria-label="Loading"
            onLoad={() => {
                if (!onLoad) return null
                return onLoad()
            }}
            className={`${customPadding || 'mt-5'}`}
            width="200"
            height="200"
            data={loadingImg}
        />
    </section>
)

export default LoadingContainer
