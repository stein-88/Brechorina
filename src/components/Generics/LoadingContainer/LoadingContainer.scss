.heigh100 {
    height: 100vh;
}