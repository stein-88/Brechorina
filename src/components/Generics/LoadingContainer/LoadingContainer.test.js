import React from 'react'
import renderer from 'react-test-renderer'
import LoadingContainer from './LoadingContainer'

describe('Test on <LoadingContainer /> component', () => {
    test('Should render without props', () => {
        const componentNoprops = renderer.create(<LoadingContainer />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
})
