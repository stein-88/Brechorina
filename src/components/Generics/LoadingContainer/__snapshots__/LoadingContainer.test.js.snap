// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <LoadingContainer /> component Should render without props 1`] = `
<section
  className="w-100 h-auto text-center pt-5 mt-5"
>
  <object
    alt="Loading"
    aria-label="Loading"
    className="mt-5"
    data="test-file-stub"
    height="200"
    onLoad={[Function]}
    width="200"
  />
</section>
`;
